### PR TITLE
Apply Webhooks CRUD into REST API

### LINKED ISSUE #17734: Update webhooks endpoints on REST API to use new Webhooks CRUD
Part of #12439

### PR DESCRIPTION
**Pull request by claudiosanches**
In order to test is necessary enable WooComerce again to create all new database tables.

Closes #17734

### CODE REVIEW

File: includes/api/v1/class-wc-rest-webhooks-controller.php [hunk 1]
```diff
@@ -133,6 +126,94 @@ public function register_routes() {
 		) );
 	}
 
+	/**
+	 * Check whether a given request has permission to read taxes.
```
Comment (claudiulodro) [inline includes/api/v1/class-wc-rest-webhooks-controller.php]: All the docblocks in this file are about taxes and they should be about webhooks. :P

**Comment by claudiulodro** [comment]
I get lots of errors when I try to create a webhook using v1 and v2 API:

v1:
```
curl --insecure -X POST https://local.wordpress.dev/wp-json/wc/v1/webhooks \
>     -u ckXXXXX:csXXXXX \
>     -H "Content-Type: application/json" \
>     -d '{
>   "name": "Order updated",
>   "topic": "order.updated",
>   "delivery_url": "http://local.wordpress.dev",
>   "secret": "XXXXX"
> }'
<br />
<b>Notice</b>:  Undefined variable: post_id in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>327</b><br />
<br />
<b>Notice</b>:  Undefined variable: data in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>331</b><br />
<br />
<b>Notice</b>:  Undefined variable: data in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>332</b><br />
<div id="error"><p class="wpdberror"><strong>WordPress database error:</strong> [Column &#039;delivery_url&#039; cannot be null]<br /><code>INSERT INTO `wp_wc_webhooks` (`status`, `name`, `user_id`, `delivery_url`, `secret`, `topic`, `date_created`, `date_created_gmt`, `api_version`, `failure_count`, `pending_delivery`) VALUES (&#039;active&#039;, &#039;Order updated&#039;, 1, NULL, &#039;o%tmV9(D,it^q[]*H5NmY7{P{=1`SG:,Tk&amp;jWTpqz^b^ %t]H,&#039;, &#039;&#039;, &#039;2017-11-30 19:59:23&#039;, &#039;2017-11-30 19:59:23&#039;, &#039;1&#039;, &#039;0&#039;, &#039;&#039;)</code></p></div><br />
<b>Fatal error</b>:  Uncaught Error: Call to a member function get_id() on null in /srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php:533
Stack trace:
#0 /srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php(349): WC_REST_Webhooks_V1_Controller-&gt;prepare_item_for_response(0, Object(WP_REST_Request))
#1 /srv/www/wordpress-default/public_html/wp-includes/rest-api/class-wp-rest-server.php(936): WC_REST_Webhooks_V1_Controller-&gt;create_item(Object(WP_REST_Request))
#2 /srv/www/wordpress-default/public_html/wp-includes/rest-api/class-wp-rest-server.php(321): WP_REST_Server-&gt;dispatch(Object(WP_REST_Request))
#3 /srv/www/wordpress-default/public_html/wp-includes/rest-api.php(266): WP_REST_Server-&gt;serve_request('/wc/v1/webhooks')
#4 /srv/www/wordpress-default/public_html/wp-includes/class-wp-hook.php(286): rest_api_loaded(Object(WP))
#5 /srv/www/wordpress-default/public_html/wp-includes/class-wp-hook in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>533</b><br />

```

v2:
```
curl --insecure -X POST https://local.wordpress.dev/wp-json/wc/v2/webhooks \
>     -u ck_xxxxx:cs_xxxxx \
>     -H "Content-Type: application/json" \
>     -d '{
>   "name": "ZOrder updated",
>   "topic": "order.updated",
>   "delivery_url": "http://local.wordpress.dev",
>   "secret": "xxxxxx"
> }'
<br />
<b>Notice</b>:  Undefined variable: post_id in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>327</b><br />
<br />
<b>Notice</b>:  Undefined variable: data in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>331</b><br />
<br />
<b>Notice</b>:  Undefined variable: data in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>332</b><br />
<div id="error"><p class="wpdberror"><strong>WordPress database error:</strong> [Column &#039;delivery_url&#039; cannot be null]<br /><code>INSERT INTO `wp_wc_webhooks` (`status`, `name`, `user_id`, `delivery_url`, `secret`, `topic`, `date_created`, `date_created_gmt`, `api_version`, `failure_count`, `pending_delivery`) VALUES (&#039;active&#039;, &#039;ZOrder updated&#039;, 1, NULL, &#039;yrD=&amp;#jpLmVv=(v#Rzi(2Du8y)PkJ*(.CJZ:F-;6yR{y^;skxM&#039;, &#039;&#039;, &#039;2017-11-30 20:01:11&#039;, &#039;2017-11-30 20:01:11&#039;, &#039;2&#039;, &#039;0&#039;, &#039;&#039;)</code></p></div>{"id":0,"name":"","status":"disabled","topic":"","resource":"","event":"","hooks":[],"delivery_url":"","date_created":null,"date_created_gmt":null,"date_modified":null,"date_modified_gmt":null,"_links":{"self":[{"href":"https:\/\/local.wordpress.dev\/wp-json\/wc\/v2\/webhooks\/0"}],"collection":[{"href":"https:\/\/local.wordpress.dev\/wp-json\/wc\/v2\/webhooks"}]}}
```

**Comment by claudiulodro** [comment]
Not necessarily a bug in this PR, but something that should probably be fixed in this PR. In this PR and master and 3.2.x, the secret is required when creating a webhook through the API (in both v1 and v2 API).

If you try and create a webhook without passing in the `secret` param you get `{"code":"rest_missing_callback_param","message":"Missing parameter(s): secret","data":{"status":400,"params":["secret"]}}`. [The docs](http://woocommerce.github.io/woocommerce-rest-api-docs/#webhook-properties) say that the `secret` param is not required and will default to an md5 hash of the user ID.

**Comment by claudiulodro** [comment]
Notice when trying to update a webhook:
v1 & v2:
```
curl --insecure -X PUT https://local.wordpress.dev/wp-json/wc/v1/webhooks/1 \
>     -u ck_b1ae8d73a6ba7543f66854920e9afcc7b0493561:cs_763339564e3650544d7a948a9c5125b4551f2a4b \
>     -H "Content-Type: application/json" \
>     -d '{
>   "status": "paused"
> }'
<br />
<b>Notice</b>:  Undefined property: stdClass::$post_title in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/api/v1/class-wc-rest-webhooks-controller.php</b> on line <b>407</b><br />
{"id":1,"name":"","status":"active","topic":"coupon.created","resource":"coupon","event":"created","hooks":["woocommerce_process_shop_coupon_meta","woocommerce_new_coupon"],"delivery_url":"http:\/\/local.wordpress.dev?v6","date_created":"2017-11-30T19:05:13","date_modified":"2017-11-30T20:14:25","_links":{"self":[{"href":"https:\/\/local.wordpress.dev\/wp-json\/wc\/v1\/webhooks\/1"}],"collection":[{"href":"https:\/\/local.wordpress.dev\/wp-json\/wc\/v1\/webhooks"}]}}
```

**Comment by claudiulodro** [comment]
501 when I try to delete a webhook:
v1 & v2:
```
curl --insecure -X DELETE https://local.wordpress.dev/wp-json/wc/v2/webhooks/1 \
>     -u ck_xxxx:cs_xxxx
{"code":"woocommerce_rest_trash_not_supported","message":"Webhooks do not support trashing.","data":{"status":501}}
```

**Comment by claudiosanches** [comment]
@claudiulodro 
> 501 when I try to delete a webhook:

This one is correct, you need to always use `?force=true` if trying to remove something that does not support trash. The same happens with other endpoints.

**Comment by claudiulodro** [comment]
Everything works great now. 👍

**Pull request merged**